person.rb

```diff
@@ -13,9 +20,36 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
     @age >= 18
   end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.length > 10 ? @nameable.correct_name[0...10] : @nameable.correct_name
+  end
+end
```

Comment:
- [ ] **[OPTIONAL]** These 3 classes should have their own files.

Reply:
Thank you @julie-ify for your suggestions!
I made sure to follow them in order to learn Ruby best practices 🙂